### PR TITLE
Api 9130 issued endpoint

### PR DESCRIPTION
**Pull request by lukemarkwordtlibertyits**
## Related PRs

https://github.com/department-of-veterans-affairs/devops/pull/9772
https://github.com/department-of-veterans-affairs/terraform-aws-dvp-oauth-proxy/pull/25
https://github.com/department-of-veterans-affairs/lighthouse-oauth-proxy-configs/pull/41

## Test Commands

**for local testing**

```sh
export HOST=http://localhost:7100/oauth2
```

**Success**

```sh
curl -s -i -X GET \
  -H "Accept: application/json" \
  -H "Authorization: Bearer 123456789" \
  $HOST/issued 
```

- [x] status 200
- [x] expected response body

```json
{
  "static": true,
  "scopes": "openid profile patient/Medication.read launch/patient offline_access",
  "expires_in": 3600,
  "icn": "555",
  "aud": "http://localhost:7100/services/static-only"
}
```
**401**

```sh
curl -s -i -X GET \
  -H "Accept: application/json" \
  -H "Authorization: Bearer bad" \
  $HOST/issued 
```

- [x] status 401
- [x] expected response body
```
unauthorized
```

## New Regression Tests functionality

```
./regression_tests.sh
```

- [x] will not run issued tests

```
# STATIC_ACCESS_TOKEN not defined
./regression_tests.sh --test-issued
```

- [x] exit 1
- [x] echo `ERROR - STATIC_ACCESS_TOKEN is a required parameter.`

```
export STATIC_ACCESS_TOKEN=123456789
./regression_tests.sh --test-issued
```

```
Running Issued Tests ...
 ✓ Valid static token
 ✓ Invalid static token
```

### CODE REVIEW

File: dev-config.base.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
   "okta_token": "okta-token",
```
Comment (crolarlibertyva) [inline dev-config.base.json:14]: Should we update the dynamo_static_token_table value to align with the local developer instance?

**Review by crolarlibertyva — APPROVED** [review]

**Review by crolarlibertyva — DISMISSED** [review]

File: src/dynamo_schema.js [hunk 2]
```diff
@@ -243,14 +260,16 @@ function createStaticTokenEntry() {
...
+      scopes: {
         S:
           "openid profile patient/Medication.read launch/patient offline_access",
       },
-      static_expires_in: { N: "3600" },
-      static_icn: { S: "555" },
+      expires_in: { N: "3600" },
+      icn: { S: "555" },
+      aud: { S: "http://localhost:7100/services/static-only" },
+      cross_hash: { S: "ada386dcfd6cbd96c0e345d0599503f488f6a7e103b1096e0bd363180204bce5" },
```
Comment (beaugrantham-va) [inline src/dynamo_schema.js]: The industry term for this is `checksum`. Might be a little clearer than `cross_hash`.

Comment (dbrown-va) [reply src/dynamo_schema.js]: Agreed, I knew there was a better name but couldn't recall.  Updated as suggested.

File: dev-config.base.json (hunk 1 above)
Comment (dbrown-va) [reply dev-config.base.json:14]: I don't think it is necessary, the dev-config.base is more to show an example of what the values could be set to, similar to the route categories and idps lists.

File: src/dynamo_schema.js [hunk 3]
```diff
@@ -243,14 +260,18 @@ function createStaticTokenEntry() {
   let itemParams = {
     TableName: "StaticTokens",
     Item: {
-      static_access_token: { S: "123456789" },
-      static_refresh_token: { S: "987654321" },
-      static_scopes: {
+      access_token: { S: "123456789" },
+      refresh_token: { S: "987654321" },
```
Comment (dbrown-va) [inline src/dynamo_schema.js:264]: I think the static refresh token flow expects this to be `static_refresh_token` still.

Comment (dbrown-va) [reply src/dynamo_schema.js:264]: Updated to use non `static_` prefix here as well as in the tests.

File: src/dynamo_schema.js [hunk 4]
```diff
@@ -243,14 +260,20 @@ function createStaticTokenEntry() {
   let itemParams = {
     TableName: "StaticTokens",
     Item: {
-      static_access_token: { S: "123456789" },
-      static_refresh_token: { S: "987654321" },
-      static_scopes: {
+      access_token: { S: "123456789" },
+      refresh_token: {
+        S: "6a9cf6b1af1d8205b771d7c7b7e1770e630f763a755b2f86833ee8ce544df25e",
```
Comment (dbrown-va) [inline src/dynamo_schema.js:265]: Note that the refresh token is hashed.

File: src/dynamo_schema.js [hunk 5]
```diff
@@ -243,14 +260,20 @@ function createStaticTokenEntry() {
...
         S:
           "openid profile patient/Medication.read launch/patient offline_access",
       },
-      static_expires_in: { N: "3600" },
-      static_icn: { S: "555" },
+      expires_in: { N: "3600" },
+      icn: { S: "555" },
+      aud: { S: "http://localhost:7100/services/static-only" },
+      checksum: {
+        S: "ada386dcfd6cbd96c0e345d0599503f488f6a7e103b1096e0bd363180204bce5",
```
Comment (dbrown-va) [inline src/dynamo_schema.js:275]: The checksum value is a combination of `{{access-token}}-{{icn}}` in addition to the hmac secret to ensure an ICN is a single value.

**Comment by dbrown-va** [comment]
A simple script to create the necessary checksum values for population in database.
https://github.com/department-of-veterans-affairs/pivot/pull/37

**Review by dbrown-va — APPROVED** [review]
Approving, but since I did a number of the latest changes would like to get at least one additional reviewer.

**Comment by lukemarkwordtlibertyits** [comment]
I can't approve because I opened this PR, but LGTM. 

A note to myself: we will have to add the `--test-issued` flag to the regression tests that we run on deployment when we enable the logic in all environments.

**Pull request merged**